core/domain/user_services.py

```diff
@@ -933,22 +976,262 @@ def create_new_user(gae_id, email):
     Raises:
         Exception. If a user with the given gae_id already exists.
     """
+    def _create_new_user_transactional(user_settings, user_auth_details):
+        """Save user models for new users as a transaction.
+
+        Args:
+            user_settings: UserSettings. The user settings domain object
+                corresponding to the newly created user.
+            user_auth_details: UserAuthDetails. The user auth details domain
+                object corresponding to the newly created user.
+        """
+        _save_user_auth_details(user_auth_details)
+        _save_user_settings(user_settings)
+        create_user_contributions(user_settings.user_id, [], [])
+
     user_settings = get_user_settings_by_gae_id(gae_id, strict=False)
     if user_settings is not None:
         raise Exception(
             'User %s already exists for gae_id %s.'
             % (user_settings.user_id, gae_id))
-
     user_id = user_models.UserSettingsModel.get_new_id('')
     user_settings = UserSettings(
         user_id, gae_id, email, feconf.ROLE_ID_EXPLORATION_EDITOR,
         preferred_language_codes=[constants.DEFAULT_LANGUAGE_CODE])
-    _save_user_auth_details(UserAuthDetails(user_id, gae_id))
-    _save_user_settings(user_settings)
-    create_user_contributions(user_id, [], [])
+    user_auth_details = UserAuthDetails(user_id, gae_id)
+    transaction_services.run_in_transaction(
+        _create_new_user_transactional, user_settings, user_auth_details)
     return user_settings
 
 
+def create_new_profiles(gae_id, email, modifiable_user_data_list):
+    """Creates new profiles for the users specified in the
+    modifiable_user_data_list.
```

Comment:
Specify whether or not this saves anything to the datastore.

Reply:
Done.